observatory_platform/telescopes/crossref_events.py

```diff
@@ -477,10 +485,11 @@ def download(**kwargs):
         start_date = pendulum.instance(kwargs['dag_run'].start_date) - timedelta(days=1)
 
         logging.info(f'Start date: {prev_start_date}, end date:{start_date}, first release: {first_release}')
-        if prev_start_date < start_date:
+        if prev_start_date > start_date:
             raise AirflowException("Start date has to be before end date.")
 
-        release = CrossrefEventsRelease(prev_start_date, start_date, first_release)
+        mail_address = BaseHook.get_connection(AirflowConn.crossref_events.get()).password
```

Comment:
If you want to use user/pass fields, may I suggest username instead for mail address. password might be harder to lookup once set due to airflow's security design